### PR TITLE
GET Todos List Method

### PR DESCRIPTION
**Pull request by LuisAlbornoz**
https://trello.com/c/RXNDSsoQ/7-crear-un-endpoint-para-listar-los-todos
We finished the task of the URL.

### CODE REVIEW

File: backend/main.py [hunk 1]
```diff
@@ -1,2 +1,49 @@
-print("Hello world!")
-print("Hello world!")
\ No newline at end of file
+from fastapi import FastAPI
+from typing import List
+from pydantic import BaseModel
+from datetime import datetime
+
+
+class Task(BaseModel):
```
Comment (Esequiel378) [inline backend/main.py:7]: Models should be on a separated file (separation of concerns)

File: backend/main.py [hunk 2]
```diff
@@ -1,2 +1,49 @@
...
+    title : str
+    color : str
+    task : List[Task]
+
+
+app = FastAPI()
+
+#create new todo
+@app.post("/todo", response_model=Item_todo)
+def create_todo(item:  Item_todo):
```
Comment (Esequiel378) [inline backend/main.py:24]: Post method is not related to this task

File: backend/main.py [hunk 3]
```diff
@@ -1,2 +1,49 @@
...
+    return item
+
+
+#Get list of todo
+@app.get("/todos", response_model=Item_todo)
+def get_todos():
+    return Item_todo (
+    id = 1,
```
Comment (Esequiel378) [inline backend/main.py:32]: This section does not meet the standard format (take a look at the black formatter)

File: backend/main.py [hunk 4]
```diff
@@ -1,2 +1,49 @@
...
+
+app = FastAPI()
+
+#create new todo
+@app.post("/todo", response_model=Item_todo)
+def create_todo(item:  Item_todo):
+    return item
+
+
+#Get list of todo
```
Comment (Esequiel378) [inline backend/main.py:28]: Comments are not documentation. Use doc strings 
https://peps.python.org/pep-0257/

**Review by Esequiel378 — CHANGES_REQUESTED** [review]
LGTM! 👌 

I left a few comments that need to be addressed before merging.

File: backend/main.py [hunk 5]
```diff
@@ -1,49 +1,29 @@
...
-        "completed" : False,
-    }
-    ]
-)
-
-
+    """Get a list of todos."""
+    return Todo(
+        id=1,
+        created_at=12 / 7 / 2007,
```
Comment (SantiiL) [inline backend/main.py:14]: for this date, is not better use datetime.now ? its better than hardcode data

Comment (ValentinMinolli) [reply backend/main.py:14]: date.time sets the real created date?

Comment (SantiiL) [reply backend/main.py:14]: datetime.now sets the current date

Comment (Esequiel378) [reply backend/main.py:14]:
```python
from datetime import datetime
datetime.now()
# datetime.datetime(2023, 2, 16, 23, 0, 36, 706646)
```

File: backend/models.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+
+class Todo(BaseModel):
+    """Class for todos."""
+
+    id: int
+    created_at: datetime
+    title: str
+    color: str
+    task: List[Task]
```
Comment (Esequiel378) [inline backend/models.py:23]: We didn't discuss what python version we will use, but we should use 3.10 or above.

If that's the case, you can use the following:

```python
task: list[Task]
```

In general, python 3.10 has better support for type hints.

File: backend/models.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
+from pydantic import BaseModel
+
+from datetime import datetime
+
+from typing import List
+
+
+class Task(BaseModel):
+    """Class for tasks."""
```
Comment (Esequiel378) [inline backend/models.py:9]: It's okay to have this documentation now, but we should create another task/spike to normalize how we will write the functions/classes documentation.

**Review by Esequiel378 — APPROVED** [review]
LGTM! 👌

File: backend/models.py (hunk 7 above)
Comment (ValentinMinolli) [reply backend/models.py:7]: So this is a future task to achieve?

Comment (Esequiel378) [reply backend/models.py:7]: That's right, we will need to plan an create a new task in trello

**Pull request merged**